relay-event-schema/src/protocol/span/convert.rs

```diff
@@ -0,0 +1,254 @@
+//! This module defines bidirectional field mappings between spans and transactions.
+use crate::protocol::{
+    ContextInner, Contexts, DefaultContext, Event, ProfileContext, Span, TraceContext,
+};
+
+use relay_protocol::Annotated;
+use std::collections::BTreeMap;
+
+/// Implements the conversion between transaction events and segment spans.
+///
+/// Invoking this macro implements both `From<&Event> for Span` and `From<&Span> for Event`.
+///
+/// Example:
+///
+/// ```skip
+/// map_fields!(
+///     top-level:
+///         span.received: event.received
+///     contexts:
+///         TraceContext:
+///             span.trace_id: context.trace_id
+///         ;
+///     ;
```

Comment:
I think I prefer 

```rust
contexts: [ 
    TraceContext {
    }
]
```

Reply:
I like the `{}` braces, so I introduced them everywhere now. Wasn't sure about `:`, `[` and `,`, so I fell back to a minimal syntax with braces.